x/ccv/provider/keeper/distribution.go

```diff
@@ -121,19 +117,36 @@ func (k Keeper) AllocateTokens(ctx sdk.Context) {
 			)
 			continue
 		}
+
+		// compute rewards for validators
+		consumerRewards := alloc.Rewards
 		voteMultiplier := math.LegacyOneDec().Sub(communityTax)
-		feeMultiplier := rewardsCollectedDec.MulDecTruncate(voteMultiplier)
+		validatorRewards := consumerRewards.MulDecTruncate(voteMultiplier)
+
+		// compute remaining rewards for the community pool
+		remaining := consumerRewards.Sub(validatorRewards)
+
+		// transfer validators rewards to distribution module account
+		validatorRewardsTrunc, validatorRewardsChange := validatorRewards.TruncateDecimal()
+		err = k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ConsumerRewardsPool, distrtypes.ModuleName, validatorRewardsTrunc)
+		if err != nil {
+			k.Logger(ctx).Error(
+				"cannot send rewards to distribution module account %s: %s",
+				consumerChainID,
+				err,
+			)
+			continue
+		}
 
 		// allocate tokens to consumer validators
-		feeAllocated := k.AllocateTokensToConsumerValidators(
+		k.AllocateTokensToConsumerValidators(
 			ctx,
 			consumerChainID,
-			feeMultiplier,
+			sdk.NewDecCoinsFromCoins(validatorRewardsTrunc...),
 		)
-		remaining = remaining.Sub(feeAllocated)
 
-		// allocate community funding
-		remainingCoins, _ := remaining.TruncateDecimal()
+		// allocate remaining rewards to the community pool
+		remainingCoins, remainingChanges := remaining.TruncateDecimal()
```

Comment:
Should this be called `remainingRewards` and `remainingChange` to be consistent with naming above.